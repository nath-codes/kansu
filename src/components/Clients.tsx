import React, { memo } from 'react'
import { rem } from 'polished'
import { animated, useTrail } from '@react-spring/web'
import styled from 'styled-components'
import { Container } from './Container'
import { Heading } from './Heading'
import { Section } from './Section'

import AmazonLogo from '../images/amazon.svg'

import Fresh8LogoL from '../images/fresh-8-l.png'
import Fresh8LogoM from '../images/fresh-8-m.png'
import Fresh8LogoS from '../images/fresh-8-s.png'
import GenomicsEnglandLogoL from '../images/genomics-england-l.png'
import GenomicsEnglandLogoM from '../images/genomics-england-m.png'
import GenomicsEnglandLogoS from '../images/genomics-england-s.png'

import MubalooLogo from '../images/mubaloo.svg'
import RHELogoL from '../images/rhe-l.png'
import RHELogoM from '../images/rhe-m.png'
import RHELogoS from '../images/rhe-s.png'
import ZoneLogo from '../images/zone.svg'

const clients = [
  { src: AmazonLogo, alt: 'Amazon logo' },
  { src: ZoneLogo, alt: 'Zone Digital logo' },
  { src: MubalooLogo, alt: 'Mubaloo logo' },
  {
    alt: 'Fresh 8 Logo logo',
    images: [Fresh8LogoS, Fresh8LogoM, Fresh8LogoL],
    src: Fresh8LogoM,
  },
  {
    alt: 'Genomics England logo',
    images: [GenomicsEnglandLogoS, GenomicsEnglandLogoM, GenomicsEnglandLogoL],
    src: GenomicsEnglandLogoM,
  },
  { src: RHELogoM, images: [RHELogoS, RHELogoM, RHELogoL], alt: 'RHE logo' },
]

const _Clients = () => {
  const trail = useTrail(6, {
    from: { opacity: 0 },
    opacity: 1,
  })

  return (
    <Section>
      <Heading>Clients</Heading>
      <Container>
        <Logos>
          {trail.map((styleProps, index) => {
            const { alt, src, images } = clients[index]

            const srcSet = images
              ? `${images[0]} 88w, ${images[1]} 184w, ${images[2]} 2x`
              : undefined
            const sizes = images ? '(min-width: 320px) 88px, 184px"' : undefined

            return (
              <StyledImg
                alt={alt}
                key={alt}
                src={src}
                sizes={sizes}
                srcSet={srcSet}
                style={styleProps}
              />
            )
          })}
        </Logos>
      </Container>
    </Section>
  )
}

const Logos = styled.div`
  align-items: center;
  display: grid;
  grid-column-gap: ${rem(24)};
  grid-row-gap: ${rem(24)};
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: ${rem(48)} ${rem(48)};

  @media only screen and (min-width: ${(props) => props.theme.breakpoints.m}) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media only screen and (min-width: ${(props) => props.theme.breakpoints.l}) {
    grid-column-gap: ${rem(48)};
    grid-row-gap: ${rem(48)};
    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: ${rem(96)} ${rem(96)};
  }
`

const StyledImg = styled(animated.img)`
  height: ${rem(48)};
  object-fit: contain;

  @media only screen and (min-width: ${(props) => props.theme.breakpoints.l}) {
    height: 100%;
  }
`

export const Clients = memo(_Clients)
