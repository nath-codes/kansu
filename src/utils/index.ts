import { WindowLocation } from 'reach__router'

const getPageNav = (location: WindowLocation | undefined, urls: string[]) => {
  if (location && location.pathname) {
    const currentIndex = urls.findIndex(
      (url: string) => url === location.pathname
    )
    const lastIndex = urls.length - 1
    const nextPage =
      currentIndex === lastIndex ? undefined : urls[currentIndex + 1]
    const prevPage = currentIndex === 0 ? undefined : urls[currentIndex - 1]

    return { nextPage, prevPage }
  }
}

export default getPageNav
