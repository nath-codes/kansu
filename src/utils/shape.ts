import {
  ANIMATION_MAX_DIMENSION,
  COLOR_DARK_BLUE_RGBA,
  COLOR_GREY_RGBA,
  COLOR_WHITE_RGBA,
} from '../constants'

const colors: [number[], number[], number[]] = [
  COLOR_DARK_BLUE_RGBA,
  COLOR_GREY_RGBA,
  COLOR_WHITE_RGBA,
]

const generateDimension = () =>
  Math.round(Math.random() * ANIMATION_MAX_DIMENSION)

const generateFill = () => {
  const fillColorKey = Math.floor(Math.random() * colors.length)
  return colors[fillColorKey]
}

const generatePosition = (limit: number, dimension: number) => {
  const edge = limit - dimension
  return Math.round(Math.random() * edge)
}

export const generateShapeFactory = (
  canvasHeight: number,
  canvasWidth: number
) => {
  const height = generateDimension()
  const width = generateDimension()

  return {
    fill: generateFill(),
    height,
    width,
    x: generatePosition(canvasWidth, width),
    y: generatePosition(canvasHeight, height),
  }
}
