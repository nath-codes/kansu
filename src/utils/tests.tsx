import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { theme } from '../styles/theme'

export const renderWithTheme = (component: JSX.Element) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
