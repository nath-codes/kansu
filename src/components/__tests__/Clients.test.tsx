import 'jest-styled-components'
import React from 'react'
import { renderWithTheme } from '../../utils/tests'
import { Clients } from '../Clients'

describe('Clients', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Clients />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
