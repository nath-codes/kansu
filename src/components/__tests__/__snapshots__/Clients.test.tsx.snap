// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clients renders correctly 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: 3rem;
}

.c1 {
  bottom: 0;
  color: #ffffff;
  font-family: "Roboto",sans-serif;
  font-size: 1.5rem;
  left: 0;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  line-height: 1;
  margin: 0;
  position: absolute;
  text-align: right;
  text-transform: uppercase;
  top: 0.75rem;
  position: absolute;
  writing-mode: sideways-lr;
}

.c1:after {
  color: #ffffff;
  content: "â– ";
  right: 0;
  top: -0.625rem;
  font-size: 0.75rem;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 3rem;
  position: relative;
}

.c0:last-child {
  margin-bottom: 3.75rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  grid-column-gap: 1.5rem;
  grid-row-gap: 1.5rem;
  grid-template-columns: repeat(2,1fr);
  grid-template-rows: 3rem 3rem;
}

.c4 {
  height: 3rem;
  object-fit: contain;
}

@media only screen and (min-width:768px) {
  .c2 {
    margin-left: 0;
    max-width: 37.5rem;
  }
}

@media only screen and (min-width:980px) {
  .c2 {
    max-width: 40.5rem;
  }
}

@media only screen and (min-width:768px) {
  .c1 {
    font-size: 1.875rem;
    margin-top: 0.625rem;
  }
}

@media only screen and (min-width:980px) {
  .c1 {
    font-size: 2.25rem;
  }

  .c1:after {
    font-size: 1.5rem;
    top: -1rem;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    margin-bottom: 4.125rem;
  }

  .c0:last-child {
    margin-bottom: 0;
    padding-bottom: 6.75rem;
  }
}

@media only screen and (min-width:768px) {
  .c3 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media only screen and (min-width:980px) {
  .c3 {
    grid-column-gap: 3rem;
    grid-row-gap: 3rem;
    grid-template-columns: repeat(3,1fr);
    grid-template-rows: 6rem 6rem;
  }
}

@media only screen and (min-width:980px) {
  .c4 {
    height: 100%;
  }
}

<section
  className="c0"
>
  <h2
    className="c1"
  >
    Clients
  </h2>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <img
        alt="Diageo logo"
        className="c4"
        src="test-file-stub"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="Guinness logo"
        className="c4"
        src="test-file-stub"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="BMI Group logo"
        className="c4"
        src="test-file-stub"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="Amazon logo"
        className="c4"
        src="test-file-stub"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="Zone Digital logo"
        className="c4"
        src="test-file-stub"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="Mubaloo logo"
        className="c4"
        src="test-file-stub"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="Fresh 8 Logo logo"
        className="c4"
        sizes="(min-width: 320px) 88px, 184px""
        src="test-file-stub"
        srcSet="test-file-stub 88w, test-file-stub 184w, test-file-stub 2x"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="Genomics England logo"
        className="c4"
        sizes="(min-width: 320px) 88px, 184px""
        src="test-file-stub"
        srcSet="test-file-stub 88w, test-file-stub 184w, test-file-stub 2x"
        style={
          {
            "opacity": 0,
          }
        }
      />
      <img
        alt="RHE logo"
        className="c4"
        sizes="(min-width: 320px) 88px, 184px""
        src="test-file-stub"
        srcSet="test-file-stub 88w, test-file-stub 184w, test-file-stub 2x"
        style={
          {
            "opacity": 0,
          }
        }
      />
    </div>
  </div>
</section>
`;
